client/system/detect_cloud/kube.go

```diff
@@ -0,0 +1,27 @@
+package detect_cloud
+
+import (
+	"os"
+	"strings"
+)
+
+func detectContainer() string {
+	b, err := os.ReadFile("/proc/self/cgroup")
```

Comment:
I believe this one doesn't work with cgroup v2 anymore, it only contains something like `0::/`. Do we want to find an alternative?